### ISSUE TITLE
link-in-text-block should consider parent nodes & descending images

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4126]
The following should pass the `link-in-text-block` rule:

```html
<p>
  Lorem ipsum 
  <u><a href="#">dolor</a></u> 
  sit amet consectetur adipisicing elit. Ea officiis
  <a href="#"><img width="16" height="16"> exercitationem</a> 
  ullam inventore facere quaerat, itaque eaque. Eaque 
  <a href="#"><svg width="16" height="16"></svg>consectetur</a>,
  doloremque nostrum, vel animi quis pariatur, qui omnis inventore cum harum!
</p>
```
